plugins/hls-hlint-plugin/src/Ide/Plugin/Hlint.hs

```diff
@@ -208,8 +216,16 @@ rules recorder plugin = do
     liftIO $ argsSettings flags
 
   action $ do
-    files <- getFilesOfInterestUntracked
-    void $ uses GetHlintDiagnostics $ Map.keys files
+    files <- Map.keys <$> getFilesOfInterestUntracked
+    Shake.ShakeExtras{ideTesting = Options.IdeTesting testing, lspEnv} <- Shake.getShakeExtras
+    let signal :: KnownSymbol s => Proxy s -> Action ()
+        signal msg = when testing $ liftIO $ Shake.mRunLspT lspEnv $
+            LSP.sendNotification (LSP.SMethod_CustomMethod msg) $
+            toJSON $ map fromNormalizedFilePath files
+
+    signal (Proxy @"kick/start/hlint")
```

Comment:
Can we do this generically? Like, for every rule, send custom method notifications for starting and finishing? It seems like many (most?) tests want this sort of thing, just having a generic signal when rules start or finish would be quite handy.

Reply:
I think only diagnostic tests need it, but yeah, we can definitely generalise it.

Reply:
I support it, since we are already tracing the action any way, see `otTracedAction`.

Reply:
We could add a log handler that intercepts rule start/finish logs and sends the custom messages, like we do for sending `window/logMessage` etc.